python/mlx/nn/layers/quantized.py

```diff
@@ -277,3 +277,120 @@ def from_linear(
             ql.bias = linear_layer.bias
 
         return ql
+
+
+class QQLinear(Module):
+    """Quantizes the input and applies an affine transformation using quantized weights.
+
+    Two use cases are supported:
+
+    1) **Inference / frozen weights**: weights are stored in quantized form together with
+       their scales (``self.weight`` is quantized and ``self.scales`` is provided).
+
+    2) **Training / weights are included in gradient computation**:
+        weights are stored in higher precision and are quantized on
+        the fly during computation.
```

Comment:
```suggestion
    2) **Train**: The weights are stored in higher precision and are quantized on
         the fly during computation so that gradients with respect to the weights
         can be computed.
```